core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -307,6 +309,20 @@ public void registerSuggester(String key, Suggester<?> suggester) {
     }
 
     /**
+     * Lookup a {@link ScoreFunctionParser} by its name as used in XContent.
+     */
+    public ScoreFunctionParser<?> lookupScoreFunctionParser(String name, ParseFieldMatcher matcher) {
+        Tuple<ParseField, ScoreFunctionParser<?>> parser = scoreFunctionParsers.get(name);
+        if (parser == null) {
+            return null;
+        }
+        if (matcher.match(name, parser.v1())) {
+            return parser.v2();
+        }
+        return null;
```

Comment:
the thing is that this is a public method which could be used anywhere else potentially. and there is never a reason why one would need to lookup a function that is not there. I think. Then why return null?